<a href="https://colab.research.google.com/github/NRJaggers/Applied-Stochastic-Processes/blob/main/PracticeProblems/CTMC_Birth_and_Death_Chains.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Birth and Death Chains

---

## Prompt

Original Prompt can be found [here](https://bookdown.org/kevin_davisross/applied-stochastic-processes/pp-ctmc-birthdeath.html). A copy of the prompt  can be found on github under [/PracticeProblems/Prompts](https://github.com/NRJaggers/Applied-Stochastic-Processes/tree/main/PracticeProblems/Prompts) along with completed exercises under [/PracticeProblems](https://github.com/NRJaggers/Applied-Stochastic-Processes/tree/main/PracticeProblems).

---

## Problems

The process below is not a birth and death chain, but a problem similar to this one can be found in the Birth and Death Chain handout.

The weather in a certain city can be in one of 3 states: sunny (1), cloudy (2), or rainy (3). Suppose the weather evolves over time according to a continuous time Markov chain with the following transition rate matrix. Rates are all per day (24 hours). (Diagonals left blank on purpose.)

$$\mathbf{Q} =
\begin{bmatrix}
     & 0.25 & 0\cr
     0.8& & 0.4\cr
     2.0& 1.5& \cr  
\end{bmatrix}$$

Given that it is currently cloudy, compute the expected number of days elapsed from now until it is sunny.

$\mu_o = \frac{0.4}{1.2}\mu_1 + \frac{0.8}{1.2}(0)$

$\mu_1 = \frac{1.5}{3.5}\mu_o + \frac{2.0}{3.5}(0)$

In [1]:
#use this to allow for running R within Python
%load_ext rpy2.ipython

In [4]:
%%R
# Followed Dr.Ross's example on absorbing states in the Birth Death Notes

QD = rbind(
  c( 0, 4) / 12,
  c(15, 0) / 35
)

# Mean amount of time spent in each state before leaving
inv_lambda = c(1 / 1.2, 1 / 3.5 )


# Solve system for mean amount of time until state 4
# See end of Handout 23

solve(diag(nrow(QD)) - QD, inv_lambda)

[1] 1.083333 0.750000


We expect 1.083 days to pass until we see a sunny day.

---

## Reflection

With each Practice Problem submission, you should also take a few minutes to reflect on the practice problems and related material and thoughtfully respond to the following.

1. Write a sentence or two summarizing - in your own words - one big idea from this material.
2. Describe one particular aspect of this material that you found challenging and why. Then describe how you dealt with the challenge, or what you can do to deal with the challenge.
3. Write at least one specific question you have about this material. I strongly encourage you to post your questions directly in Discord as that's where I'll respond to questions. That is, if you really want an answer to your question - which I hope you do! - please post the question directly in Discord.

I understand this section is suppose to relate what we have learned about CTMC to Birth-Death Chains. Here we practice the absorbing technique discussed at the end of this sections lecture notes (I hope that was the right approach). It took some time but I think I got it. Best to probably check in class, OH or discord.